datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTableTypeRegistry.java

```diff
@@ -57,6 +58,44 @@ public DataTableTypeRegistry(Locale locale) {
         TableCellTransformer<Double> doubleTableCellTransformer = applyIfPresent(numberParser::parseDouble);
         defineDataTableType(new DataTableType(Double.class, doubleTableCellTransformer));
         defineDataTableType(new DataTableType(double.class, doubleTableCellTransformer));
+
+        TableCellTransformer<Optional<Object>> optionalTableCellTransformer =
+                (String s) -> s == null || s.isEmpty() ? Optional.empty() : Optional.of(s);
```

Comment:
The pattern `s == null || s.isEmpty() ? Optional.empty() : Optional.of(do-something-transformer-specific)` is very repetitive. Have a look at the way `applyIfPresent` is used to see if you can remove this repetition this..